internal/engine/upper_test.go

```diff
@@ -3630,6 +3660,15 @@ func (f *testFixture) nextCall(msgAndArgs ...interface{}) buildAndDeployCall {
 	}
 }
 
+func (f *testFixture) imageTargsForNextNCalls(n int, msgAndArgs ...interface{}) []model.ImageTarget {
```

Comment:
this helper function seems problematic.

It invents a new way to assert things about calls, rather than using the existing assert conventions. And it bakes in some assumptions that aren't true in all test cases (1 ImageTarget per Manifest), then hides that assumption.

Reply:
i can fix the 1 target per manifest thing for sure -- but i need some order-agnostic way of checking calls, which is what this attempted to be. let me see if i can make it cleaner.